src/cmap/mod.rs

```diff
@@ -176,20 +176,16 @@ impl ConnectionPool {
 
     /// Increments the generation of the pool. Rather than eagerly removing stale connections from
     /// the pool, they are left for the background thread to clean up.
-    pub(crate) fn clear(&self) {
-        self.manager.clear();
+    pub(crate) async fn clear(&self) {
+        self.manager.clear().await
     }
 
     /// Mark the pool as "ready", allowing connections to be created and checked out.
     pub(crate) async fn mark_as_ready(&self) {
         self.manager.mark_as_ready().await;
     }
 
-    /// Subscribe to updates to the pool's generation.
-    ///
-    /// This can be used to listen for errors that occur during connection
-    /// establishment or to get the current generation of the pool.
-    pub(crate) fn subscribe_to_generation_updates(&self) -> PoolGenerationSubscriber {
-        self.generation_subscriber.clone()
+    pub(crate) fn generation(&self) -> u32 {
```

Comment:
Since `clear` is async now, we no longer need to subscribe to updates and can just `await` on the clear. We still need to be able to get the generation at a point in time though.